profiling/src/profile/internal/observation/observations.rs

```diff
@@ -50,31 +53,31 @@ impl Observations {
                 .for_each(|(a, b)| *a += b);
         } else {
             let trimmed = TrimmedObservation::new(values, obs_len);
-            observations.aggregated_data.insert(sample_id, trimmed);
+            observations.aggregated_data.insert(sample, trimmed);
         }
     }
 
     pub fn is_empty(&self) -> bool {
         self.inner.is_none()
     }
 
-    pub fn iter(&self) -> impl Iterator<Item = (SampleId, Option<Timestamp>, &[i64])> {
+    pub fn iter(&self) -> impl Iterator<Item = (&Sample, Option<Timestamp>, &[i64])> {
```

Comment:
Since `Sample` is only 64-bit, I think it would be nicer to return a `Sample` instead of a `&Sample`, particularly because this is immutable.

Reply:
Done